packages/firestore/src/local/simple_db.ts

```diff
@@ -589,6 +589,25 @@ export class SimpleDbStore<
     keyOrValue: KeyType | ValueType,
     value?: ValueType
   ): PersistencePromise<void> {
+    try {
+      keyOrValue =
+        keyOrValue != null
+          ? JSON.parse(JSON.stringify(keyOrValue))
```

Comment:
This would mean we do this for every transaction we do. We need to think about implications (performance etc).

Reply:
Thanks for the review! You're correct, I'm concerned about the performance impact as well. And this definitely doesn't fix the underlying issue (functions being added into the indexedDB), it's only a band-aid for those who desperately need a solution.

The firebase team should definitely consider a better solution for the final patch!

Reply:
Can you explain why a function is being written to indexed db or why it only happens to Vue, if you know the answers?

Reply:
I couldn't determine *why* functions were being added, and why that was happening only in Vue3. But I was able to see *what* functions/Objects were being added:
<img width="501" alt="db-error-reason" src="https://user-images.githubusercontent.com/20404478/166627518-754ec56b-b860-4ac0-aed7-e4ff8563323a.png">
The culprit here seems to be the document->fields->**Proxy** Object. Firestore is attempting to insert this Proxy object into the IndexedDB and that's causing the error.

Reply:
I _think_ this is not added by our codebase ([reference](https://cs.opensource.google/search?q=Proxy&sq=&ss=firebase-sdk%2Ffirebase-js-sdk:packages%2Ffirestore%2F)).

Reply:
That's very likely. Would also explain why this is only happening on Vue3. Possibly in one of the shared libraries for reactivity. Will investigate more and see if I can find something.